src/test/java/bio/terra/service/auth/iam/IamServiceTest.java

```diff
@@ -305,4 +306,45 @@ void testRemoveEmailFromGroup() throws InterruptedException {
         .thenReturn(List.of());
     assertEquals(iamService.removeEmailFromGroup(groupName, policyName, email), List.of());
   }
+
+  @Test
+  void setResourceParent() throws InterruptedException {
+    UUID childId = UUID.randomUUID();
+    UUID parentId = UUID.randomUUID();
+    iamService.setResourceParent(
+        TEST_USER.getToken(),
+        IamResourceType.DATASNAPSHOT,
+        childId,
+        IamResourceType.DATASET,
+        parentId);
+    verify(iamProvider)
+        .setResourceParent(
+            TEST_USER.getToken(),
+            IamResourceType.DATASNAPSHOT,
+            childId,
+            IamResourceType.DATASET,
+            parentId);
+  }
+
+  @Test
+  void deleteResourceParent() throws InterruptedException {
+    UUID childId = UUID.randomUUID();
+    iamService.deleteResourceParent(TEST_USER.getToken(), IamResourceType.DATASNAPSHOT, childId);
+    verify(iamProvider)
+        .deleteResourceParent(TEST_USER.getToken(), IamResourceType.DATASNAPSHOT, childId);
+  }
+
+  @Test
+  void getResourceParent() throws InterruptedException {
+    UUID childId = UUID.randomUUID();
+    FullyQualifiedResourceId parent =
+        new FullyQualifiedResourceId()
+            .resourceTypeName(IamResourceType.DATASET.getSamResourceName())
+            .resourceId(UUID.randomUUID().toString());
```

Comment:
This is fine but it the contents of `parent` are unused in the code being tested so these lines have no effect on the test outcome. The only thing tested here is that the IamService API is returning the object that the IamProvider is returning.
```suggestion
    FullyQualifiedResourceId parent = new FullyQualifiedResourceId();
```